### ISSUE TITLE
Issue with Add-OVStorageSystem

### ISSUE THREAD

**Issue by oolivera15** [HewlettPackard/POSH-HPOneView#603]
**Expected Behavior**
Add-OVStorageSystem should support Alletra9000.

**Actual Behavior**
Add-OVStorageSystem does not support Alletra9000.

**Steps to reproduce**
```
$hostname                           = "<alletra9000_ip>"
$family                             = "Alletra9000"
Add-OVStorageSystem -Hostname $hostName -Username <user> -Password <password> -Family $family
Add-OVStorageSystem: Cannot validate argument on parameter 'Family'. The argument "Alletra9000" does not belong to the set "StoreServ,StoreVirtual,Nimble,Primera" specified by the ValidateSet attribute. Supply an argument that is in the set and then try the command again.
```

**Version Information**
Powershell:

```
> Get-Host | Select-Object Version

Version
-------
7.0.4
```
HPEOneView: 6.50.00-0452161

HPEOneView module:

```
> (Get-Module -Name HPEOneView.630).Version

Major  Minor  Build  Revision
-----  -----  -----  --------
6      30     2928   3678
```
Could you please help us to solve this issue?
Thank you!

**Comment by oolivera15** [comment]
Hello!

Could someone please help us with this issue? We need a solution urgently.

Thank you!

**Comment by jullienl** [comment]
Until the library is updated to support Alletra 9000, you can make a direct call to the Oneview API using `invoke-webrequest`. You will need to build your script like https://github.com/jullienl/HPE-Synergy-OneView-demos/blob/master/Powershell/OneView/Disable-OneView-TLS1.1.ps1 but using :
```
POST /rest/storage-systems
{
    "hostname" : "example.com",
    "username" : "username",
    "password" : "password",
    "family" : "Alletra9000"
    
}
```

**Comment by oolivera15** [comment]
Hi Jullien!

Thank you for your answer.
That procedure allows to add Alletra9000 in "Connected" state, but it not possible to change it to "Managed".
Is there any way to add Alletra9000 in OneView in "Managed" state?

Thank you!

**Comment by jullienl** [comment]
I think you just need to perform a GET on your storage system using `GET /rest/storage-systems/{id}` and use the storage system object to create a new request body but this time using the `PUT` method and changing the `state` property to `Managed`.

**Comment by oolivera15** [comment]
Hi Jullien!

I have tried the procedure you described:
1-Add Alletra9000 in "Connected" state using POST /rest/storage-systems
2-Get Alletra9000 object using GET /rest/storage-systems/{id} and modify state to "Managed"
3-Try to modify Alletra9000 to "Connected" state using PUT /rest/storage-systems/{id}

However it seems that state property cannot be changed once Alletra9000 has been added, it cannot be modified in OneView GUI. Step 3 give me the following error:
`Invoke-WebRequest: {"errorSource":null,"nestedErrors":[],"errorCode":"STRM_BAD_REQUEST","data":{},"recommendedActions":["Use the current value of the property in the request. In some cases, read-only properties can be omitted from the request or set to null in order to ignore changes that may have occurred to the value."],"details":null,"message":"Unable to modify the value for read-only property 'state'."}`

Since it seems that state property must be set before adding Alletra9000, I have followed theses steps:
1-Add Alletra9000 in "Connected" state using POST /rest/storage-systems
2-Get Alletra9000 object using GET /rest/storage-systems/{id} and modify state to "Managed"
3-Remove Alletra9000 from OneView GUI
4-Add Alletra9000 using POST /rest/storage-systems and the modified object got in step 2.

This procedure doesn´t work either, the following error is prompted:
`Invoke-WebRequest: {"errorSource":null,"nestedErrors":[],"errorCode":"REQUEST_DATA_READ_ERROR","data":{},"recommendedActions":["Refer to the API specification to determine the appropriate JSON request body for this request, then try the request again."],"details":"There was a problem near uri at line 371, position 8.","message":"Unable to parse JSON request body."}`

Any other ideas?
Thank you!

**Comment by jullienl** [comment]
You can perform the add storage operation using the OneView GUI by pressing F12 to open the developer tools so you can inspect in the Network tab the calls used through the browser. Locate the PUT and POST calls and examine the Request/Payload sub-tab to see the body content you need to build. Then use the same con tent in your PowerShell script.

**Comment by ChrisLynchHPE** [comment]
Support is being added to the 6.30 library.

**Comment by ChrisLynchHPE** [comment]
This is fixed in [Release 6.30.3188.1900](https://github.com/HewlettPackard/POSH-HPEOneView/releases/tag/v6.30.3188.1900) and in [PowerShell Gallery](https://www.powershellgallery.com/packages/HPEOneView.630/6.30.3188.1900).

Please verify before this issue is closed.

**Comment by ChrisLynchHPE** [comment]
Closing due to no further activity.  If you wish, feel free to re-open to discuss further.
